### PR TITLE
Automatic ChainerX fallback of array attributes in `Function` 

### PR DESCRIPTION
**Pull request by niboshi**
ChainerX array attribute fallback for `chainer.Function` similar to #5745

### CODE REVIEW

File: chainer/function_node.py [hunk 1]
```diff
@@ -146,7 +147,11 @@ class FunctionNode(object):
     _retained_output_data = None
     _local_function_hooks = None
     _supports_static_optimizations = False
-    _is_chainerx = None
+    # True if the function node is operating on ChainerX arrays and it falls
+    # back to NumPy/CuPy implementation.
+    _is_chainerx_fallback = False
```
Comment (asi1024) [inline chainer/function_node.py]: In this implementation, `False` will be set to this variable whenever `is_chainerx` is `False` (even if chainerx implementation of the function is not supported).  I think it seems a little bit confusing.  How about assign `None` or anything like that which means "unknown state" in this line (or delete this line)?

Comment (niboshi) [reply chainer/function_node.py]: Renamed to `_is_chainerx_fallback_mode` as discussed offline

**Comment by asi1024** [comment]
jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**